Create and add Table of Contents to about page

safwanyp opened the pull request:
Closes #200 

---

I did not want to touch any of the existing code outside the `about` page, which is why one might notice some repeated styles. I did this to make sure that even if this PR is merged, nothing other than the about page is affected. 

I ran `pnpm audit --fix` to address the vulnerabilities, but if it seems unnecessary I can get rid of that commit. 

I noticed guidelines for [Copyright Headers](https://github.com/opensourcepledge/opensourcepledge.com/blob/862c9b63c7ab16edea29fb64e8eca22b9ec1274f/CONTRIBUTING.md#copyright-headers) in the contributing guidelines but wasn't sure if I should add anything. If I need to, please do let me know 🙏🏻

Feedback is appreciated!

vladh commented:
Thank you @safwanyp! I'll review this soon. Just a quick note: you should definitely add copyright headers.

chadwhitacre commented:
I think this is a net positive, thanks @safwanyp. I would make it always shown (no collapse) and no border on it, and probably an h2 or h3 for the "Table of Contents" headers. It looks like the anchors are manual. Maybe a follow-up to this (separate PR) would be to make them dynamic.

<img width="965" alt="image" src="https://github.com/user-attachments/assets/282af9b8-5b1b-4631-b5ef-16fc70d801aa">

safwanyp commented:
Noted! How does this look? Will push a commit if it looks good 😄 
![osp-sc](https://github.com/user-attachments/assets/8e3246bc-1964-4580-9ecf-013d9414e78d)

chadwhitacre commented:
Left looks good. What is the right screenshot showing?

safwanyp commented:
> Left looks good. What is the right screenshot showing?

Right is the desktop (big screen) view. Do you want the big screen to not have the changes you mentioned?

chadwhitacre commented:
Ah, seeing it now. Looks good. 👍

safwanyp commented:
- made the change suggested by @chadwhitacre 
- rebased on origin/main to resolve conflicts without an extra commit
- force-pushed due to rebase

chadwhitacre commented:
Hrm ... are the pnpm lockfile and npm upgrades strictly necessary?

safwanyp commented:
> Hrm ... are the pnpm lockfile and npm upgrades strictly necessary?

Those were strictly due to me running `pnpm audit --fix` (it's part of my install alias 😅). If you prefer, I can just remove that commit completely

chadwhitacre commented:
Yes please, `+4,820 −567` is a scary diff. :)

safwanyp commented:
Yeah i think I made it worse by using `pnpm` instead of `npm` 😂 
Anyways, I've removed that commit completely, and force pushed.

chadwhitacre commented:
Thanks ... now seeing this detail: the blob is behind the TOC on wide screen, kinda unideal for it to get cutoff, but presumably it doesn't look great if the TOC background is transparent. 🤔

<img width="506" alt="image" src="https://github.com/user-attachments/assets/ddfa943a-d3c1-425b-ad69-0568bbf9ef70">

chadwhitacre commented:
🤔🤔🤔🤔🤔

<img width="421" alt="image" src="https://github.com/user-attachments/assets/946e542e-60a5-46a7-b857-0ad63918aa52">

safwanyp commented:
Yes the issue was that background being transparent made the contents hard to read, when over a blob. This is my opinion, but if you think it looks good, I can quickly change the styling!

chadwhitacre commented:
Also this is more of an edge case but at wide width and short height the TOC is cutoff with no way to access lower items.

<img width="1620" alt="image" src="https://github.com/user-attachments/assets/07672d54-5965-48ad-aff6-6e66d505e75c">

I'm leaning towards using the narrow-width layout universally. Wdyt?

chadwhitacre commented:
> If I understand correctly, you're leaning towards having the TOCs at the top always?

Yes. Seem reasonable to get this done w/o getting bogged down in design details?

> I'll be away for the rest of the night (CET).

👍 Thanks for tackling this.

safwanyp commented:
> > If I understand correctly, you're leaning towards having the TOCs at the top always?
> 
> Yes. Seem reasonable to get this done w/o getting bogged down in design details?

Yeah that sounds completely valid and fair. I'll push a commit soon

safwanyp commented:
Hey @chadwhitacre @vladh 
I've pushed all the latest changes. Please have a look and let me know what you think!

Brief summary of latest changes:
- I've moved the table of contents to the top of the page (under the page heading) across all views, to make sure it's scoped to the actual content.
- I've removed all the duplicated classes as they were only needed to make sure the "floating" worked.

chadwhitacre commented on src/pages/about.astro:
```diff
@@ -25,254 +27,256 @@ const members = filterInactiveMembers(await getMembers());
+        </section>
+
+        <section>
+          <h2>What does joining the Pledge entail?</h2>
```
Is there a reason not all FAQs are included in the TOC?

safwanyp replied:
Ah shoot I think i messed up when resolving merge conflicts. will push a fix for this tomorrow!

safwanyp replied:
Pushed a commit for this!

safwanyp commented:
@vladh Appreciate the comments, will definitely help in the future 😁

chadwhitacre commented:
Thanks for the contribution, @safwanyp!
